pkg/util/hostname.go

```diff
@@ -122,9 +137,14 @@ func GetHostname() (string, error) {
 	configName := config.Datadog.GetString("hostname")
 	err = ValidHostname(configName)
 	if err == nil {
-		cache.Cache.Set(cacheHostnameKey, configName, cache.NoExpiration)
-		setHostnameProvider("configuration")
-		return configName, err
+		provider = HostnameProviderConfiguration
+		hostnameData := HostnameData{Hostname: configName, Provider: provider}
+		cache.Cache.Set(cacheHostnameKey, hostnameData, cache.NoExpiration)
+		setHostnameProvider(provider)
+		if !checkIfHostnameUsedAsCanonicalHostname(configName) {
+			_ = log.Warnf("Hostname %s will not be used. Please see https://github.com/DataDog/documentation/blob/master/content/en/agent/faq/hostname-starting-with-ec2-default-prefix.md for more information", configName)
```

Comment:
let's create a short bitly link for this. If the location of the docs changes in the future, that'll allow us to update what the bitly link points to. Ping me on slack for how to create a link

Also, let's reword this a bit: `Hostname '%s' defined in configuration will not be used as the in-app hostname. [...]`

Reply:
Done.